ReactAndroid/src/main/java/com/facebook/react/views/scroll/ReactHorizontalScrollView.java

```diff
@@ -423,12 +426,57 @@ protected void onScrollChanged(int x, int y, int oldX, int oldY) {
     }
   }
 
+  private HorizontalScrollView findNestedScrollViewForMotionEvent(MotionEvent ev) {
+    Queue<ViewGroup> viewsToSearch = new LinkedList<>();
```

Comment:
Although using Queue is fine here for BFS, I believe we can use recursion instead to implement DFS. It will avoid extra allocations (and probably going to be a bit faster)